src/components/tl/mlx5/alltoall/alltoall_coll.c

```diff
@@ -752,26 +783,44 @@ static inline int power2(int value)
     return p;
 }
 
-static inline int block_size_fits(size_t msgsize, int block_size)
+static inline int block_size_fits(size_t msgsize, int height, int width)
 {
-    return block_size * ucc_max(power2(block_size) * msgsize, MAX_MSG_SIZE) <=
-           MAX_TRANSPOSE_SIZE;
+    size_t t1    = power2(ucc_max(msgsize, 8));
+    size_t tsize = height * ucc_max(power2(width) * t1, MAX_MSG_SIZE);
+
+    return tsize <= MAX_TRANSPOSE_SIZE && msgsize <= 128 && height <= 64 &&
```

Comment:
define literals and not hardcode

Reply:
Are you talking about the 128 and 64? If so, yes agreed.

Reply:
yes